service.py

```diff
@@ -0,0 +1,85 @@
+# the module contains the client and server process to manage ids
+
+import logging
+import os
+import json
+import time
+
+os.umask(0o007)
+
+# we avoid the GIL by using a process to actually keep track of the ids
+from multiprocessing.connection import Listener, Client
+
+import secrets
+
+logger = logging.getLogger(__name__)
+
+auth_key = secrets.token_bytes(16)
+
+address = None
+
+directory_root = None
+
+ids = {}
+
+# load_ids reads the file system for all to keep a cache of ids in memory
+def load_ids(directory):
+    for root, sub_dirs, files in os.walk(directory):
+        for file_name in files:
+            if file_name.endswith('.data'):
+                (code, date, extension) = file_name.split('.')
```

Comment:
Is putting the date in the filename better than using the date of the file.  

I'm thinking of multis-server case, when a user ask (/green) for data since a date, its going to want all date that arrived at this server since "date" as some data might have come from elsewhere and take a while to arrive ? (Can explain on call if this isn't clear (I'm finding it hard to phrase)

Reply:
yes, this reduces the need to do a stat on the file system.  My forward looking thought is that i will use some time ordered data structure internally to do that.  Of course other tools could also do a ``find --since`` filesystem command.

Reply:
I think what you've chosen is fine - it could actually be better since there may be multiple reports for a user.